core/sail/api/src/main/java/org/eclipse/rdf4j/sail/helpers/AbstractNotifyingSailConnection.java

```diff
@@ -48,16 +48,16 @@ protected boolean hasConnectionListeners() {
 		return !listeners.isEmpty();
 	}
 
-	protected void notifyStatementAdded(Statement st) {
+	protected void notifyStatementAdded(Statement st, boolean inferred) {
```

Comment:
This method is protected, so it can be overridden by any subclass. That makes this an API level change, since the existing method signature is removed. 

You can try to keep the existing method and have it call the new method, probably best to assume that all statements are explicit for backwards compatibility.